cluster_pack/uploader.py

```diff
@@ -251,29 +251,43 @@ def _upload_env_from_venv(
 
     _logger.debug(f"Packaging current_packages={reqs}")
 
-    if force_upload or not _is_archive_up_to_date(package_path, reqs, resolved_fs):
-        _logger.info(
-            f"Zipping and uploading your env to {package_path}"
-        )
-
-        if include_editable:
-            editable_requirements = packaging.get_editable_requirements()
-        else:
-            editable_requirements = {}
+    if not force_upload and _is_archive_up_to_date(package_path, reqs, resolved_fs):
+        _logger.info(f"{package_path} already exists")
+        return
 
-        with tempfile.TemporaryDirectory() as tempdir:
+    with tempfile.TemporaryDirectory() as tempdir:
+        env_copied_from_fallback_location = False
+        temp_env = f'{tempdir}/{packer.env_name()}.{packer.extension()}'
+
+        if not force_upload and 'ENV_FALLBACK_PATH' in os.environ:
+            fs_scheme = packaging.get_default_fs()
+            filename = f'{packer.env_name()}.{packer.extension()}'
+            fallback_package_path = f"{fs_scheme}/{os.environ['ENV_FALLBACK_PATH']}/{filename}"

```

Comment:
I would use only ENV_FALLBACK_PATH and pass the full pex path with fs_scheme sth like:
ENV_FALLBACK_PATH=viewfs://user/myuser/mexfile.pex

The target path will be determined by temp_env